### DISCUSSION TITLE
Facing issue in calling of data fetcher method

### DISCUSSION THREAD

**Discussion by manu-mnnit** [Netflix/dgs-framework#241] [Q&A]
Hi,

I am newbie in JAVA and graphQL.

I was using DGS framework demo example i.e. Shows Example.

My project is getting compiled and runs successfully, but Query is returning NULL everytime.
Query ::
{
    shows {
        title
        releaseYear
    }
}

OR
{
    shows(titleFilter: "Ozark") {
        title
    }
}

I am using JAVA 1.8, Spring 2.4.4, and DGS version 3.10.2. I have modofied showDataFetcher.java as per 1.8 compatibility.

Any help will be appreciated.

**Comment by paulbakker** [comment]
That probably means your datafetcher wasn't found. Can you post your datafetcher class?

**Reply by manu-mnnit** [reply]
Hi @paulbakker,

One more thing, i recently came to know that i have only JRE1.8 in my machine not JDK8. although project was compiling with few errors and running ok. so could it be reason for no call to endpoint method.

This is data fetcher class, modified shows list initialization a bit to work with java 1.8.

@DgsComponent
public class ShowsDatafetcher {

    private final List<Show> shows = Arrays.asList(
            new Show("Stranger Things", 2016),
            new Show("Ozark", 2017),
            new Show("The Crown", 2016),
            new Show("Dead to Me", 2019),
            new Show("Orange is the New Black", 2013)
    );

    @DgsData(parentType = "Query", field = "shows")
    public List<Show> shows(@InputArgument("titleFilter") String titleFilter) {
        if(titleFilter == null) {
            return shows;
        }

        return shows.stream().filter(s -> s.getTitle().contains(titleFilter)).collect(Collectors.toList());
    }
}

public class Show {
    private final String title;
    private final Integer releaseYear   ;

    public Show(String title, Integer releaseYear) {
        this.title = title;
        this.releaseYear = releaseYear;
    }

    public String getTitle() {
        return title;
    }

    public Integer getReleaseYear() {
        return releaseYear;
    }
}

**Reply by manu-mnnit** [reply]
Hi @paulbakker,

i got it running, it was possibly failing due to missing JDK1.8. My machine had only jre1.8.

Thank you for your anticipation.
